Sources/wiretapp/WiretappURLProtocol.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+
+private var urlCounter: [String: Int] = [:]
+
+public class WiretappURLProtocol: URLProtocol {
+    typealias Output = (data: Data, response: URLResponse)
+    public override class func canonicalRequest(for request: URLRequest) -> URLRequest {
+        return request
+    }
+
+    public override class func canInit(with request: URLRequest) -> Bool {
+        ProcessInfo.processInfo.environment[Wiretapp.testCaseName] != nil
+    }
+
+    public override func startLoading() {
+        let processedInfo = send(request)
+        switch processedInfo {
+        case .success(let output):
+            client?.urlProtocol(self, didReceive: output.response, cacheStoragePolicy: .notAllowed)
+            client?.urlProtocol(self, didLoad: output.data)
+
+        case .failure(let error):
+            fatalError(error.localizedDescription)
+        }
+        client?.urlProtocolDidFinishLoading(self)
+    }
+
+    public override func stopLoading() {}
```

Comment:
should we store a cancellable so if this gets called we can stop the request? I suppose since its loading from disk this should not really be an issue?